cmd/kubeadm/app/cmd/upgrade/apply.go

```diff
@@ -100,39 +105,63 @@ func newCmdApply(apf *applyPlanFlags) *cobra.Command {
 //   - Uploads the newly used configuration to the cluster ConfigMap
 //   - Creating the RBAC rules for the bootstrap tokens and the cluster-info ConfigMap
 //   - Applying new CoreDNS and kube-proxy manifests
-func runApply(flags *applyFlags, args []string) error {
+func runApply(flagSet *pflag.FlagSet, flags *applyFlags, args []string) error {
 
 	// Start with the basics, verify that the cluster is healthy and get the configuration from the cluster (using the ConfigMap)
 	klog.V(1).Infoln("[upgrade/apply] verifying health of cluster")
 	klog.V(1).Infoln("[upgrade/apply] retrieving configuration from cluster")
-	client, versionGetter, cfg, err := enforceRequirements(flags.applyPlanFlags, args, flags.dryRun, true, &output.TextPrinter{}, loadConfig)
+	client, versionGetter, initCfg, upgradeCfg, err := enforceRequirements(flagSet, flags.applyPlanFlags, args, flags.dryRun, true, &output.TextPrinter{})
 	if err != nil {
 		return err
 	}
 
 	// Validate requested and validate actual version
 	klog.V(1).Infoln("[upgrade/apply] validating requested and actual version")
-	if err := configutil.NormalizeKubernetesVersion(&cfg.ClusterConfiguration); err != nil {
+	if err := configutil.NormalizeKubernetesVersion(&initCfg.ClusterConfiguration); err != nil {
 		return err
 	}
 
 	// Use normalized version string in all following code.
-	newK8sVersion, err := version.ParseSemantic(cfg.KubernetesVersion)
+	newK8sVersion, err := version.ParseSemantic(initCfg.KubernetesVersion)
 	if err != nil {
-		return errors.Errorf("unable to parse normalized version %q as a semantic version", cfg.KubernetesVersion)
+		return errors.Errorf("unable to parse normalized version %q as a semantic version", initCfg.KubernetesVersion)
 	}
 
-	if err := features.ValidateVersion(features.InitFeatureGates, cfg.FeatureGates, cfg.KubernetesVersion); err != nil {
+	if err := features.ValidateVersion(features.InitFeatureGates, initCfg.FeatureGates, initCfg.KubernetesVersion); err != nil {
 		return err
 	}
 
 	// Enforce the version skew policies
 	klog.V(1).Infoln("[upgrade/version] enforcing version skew policies")
-	if err := EnforceVersionPolicies(cfg.KubernetesVersion, newK8sVersion, flags, versionGetter); err != nil {
+	allowRCUpgrades, ok := cmdutil.ValueFromFlagsOrConfig(flagSet, options.AllowRCUpgrades, upgradeCfg.Apply.AllowRCUpgrades, &flags.allowRCUpgrades).(*bool)
+	if ok {
+		flags.allowRCUpgrades = *allowRCUpgrades
+	} else {
+		return errors.New("type mismatch, AllowRCUpgrades is expected to be a pointer to bool")
```

Comment:
a helper function can be used, maybe in common.go, so that the same error string is not copied many times?

```
getBoolPtrError := func(opt string) error {
	return errors.Errorf("type mismatch, %s is expected to be a pointer to bool", opt)	
}

```

Reply:
Done